### PR TITLE
Create a runnable Jenkinsfile Runner demo.

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This is a copy of a more complicated CWP demo, but I believe it’s useful to have something in this repo as well.

### CODE REVIEW

**Review by jvz — APPROVED** [review]

File: demo/cwp/Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Just a Makefile for manual testing
+.PHONY: all
```
Comment (alecharp) [inline demo/cwp/Makefile:2]: `clean` and `run` should also be part of `PHONY`.

File: demo/cwp/Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ARTIFACT_ID = jenkinsfile-runner-demo
+VERSION = 256.0-test
+CWP_VERSION = 1.3
+
+all: clean build
+
+clean:
+	rm -rf tmp
+
+.build/cwp-cli-${CWP_VERSION}.jar:
```
Comment (alecharp) [inline demo/cwp/Makefile:13]: why did you prefixed it with a `.`? If this is a real file, then don't, but add `build` in `PHONY` listing.

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@ bootstrap/target/
 payload-dependencies/target/
 payload/target/
 setup/target/
+cwp-cli-1.3.jar
```
Comment (alecharp) [inline .gitignore:6]: From the `Makefile`, shouldn't this file be in `build`?

Comment (oleg-nenashev) [reply .gitignore:6]: Yes, historical stuff

**Pull request merged**